oxenc/bt_producer.h

```diff
@@ -308,6 +314,15 @@ class bt_list_producer {
         append_intermediate_ends();
     }
 
+    /// Appends an element containing const_span data
+    template <const_span_like T>
+    void append(const T& data) {
```

Comment:
There's no need to restrict this to const spans, but there *is* a need to restrict this to basic_char spans:
```suggestion
    template <basic_char T>
    void append(const std::span<T>& data) {
```